internal/backend/triggers/triggers.go

```diff
@@ -3,106 +3,114 @@ package triggers
 )
 
-// NOTE(s):
-// 1. atomicity. Create temporal schedule and save trigger aren't atomic (and not temporal workflow).
-//    It's possible that AK will crash after creating schedule but before saving trigger. In this case
-//    the schedule will be orphaned. It will continue to be triggered by temporal, but will fail to run,
-//    since it needs to fetch the trigger form DB.
-//    We need to decide how to handle this case
-// 2. Same note on any non-atomic schedule creation and DB/trigger update
-
 type triggers struct {
-	z         *zap.Logger
+	sl        *zap.SugaredLogger
 	db        db.DB
-	scheduler sdkservices.Scheduler
+	scheduler *scheduler.Scheduler
 }
 
-func New(z *zap.Logger, db db.DB, tsc sdkservices.Scheduler) sdkservices.Triggers {
-	return &triggers{db: db, z: z, scheduler: tsc}
+func New(l *zap.Logger, db db.DB, scheduler *scheduler.Scheduler) sdkservices.Triggers {
+	return &triggers{db: db, sl: l.Sugar(), scheduler: scheduler}
 }
 
 func (m *triggers) Create(ctx context.Context, trigger sdktypes.Trigger) (sdktypes.TriggerID, error) {
 	if trigger.ID().IsValid() {
 		return sdktypes.InvalidTriggerID, errors.New("trigger id already defined")
 	}
 
+	if trigger.WebhookSlug() != "" {
+		return sdktypes.InvalidTriggerID, sdkerrors.NewInvalidArgumentError("webhook slug cannot be set")
+	}
+
 	trigger = trigger.WithNewID()
 
-	if trigger.ConnectionID() == sdktypes.BuiltinSchedulerConnectionID {
-		if schedule, _ := trigger.Data()[fixtures.ScheduleExpression].ToString(); schedule != "" {
-			if err := m.scheduler.Create(ctx, trigger.ID().String(), schedule, trigger.ID()); err != nil {
-				return sdktypes.InvalidTriggerID, err
-			}
+	sl := m.sl.With("trigger_id", trigger.ID())
+
+	switch trigger.SourceType() {
+	case sdktypes.TriggerSourceTypeWebhook:
+		trigger = webhookssvc.InitTrigger(trigger)
+		sl.With("slug", trigger.WebhookSlug()).Infof("creating webhook trigger with slug %q", trigger.WebhookSlug())
+	case sdktypes.TriggerSourceTypeSchedule:
+		if err := m.scheduler.Create(ctx, trigger.ID(), trigger.Schedule()); err != nil {
+			return sdktypes.InvalidTriggerID, fmt.Errorf("create schedule: %w", err)
 		}
+		sl.With("schedule", trigger.Schedule()).Infof("creating schedule trigger with spec %q", trigger.Schedule())
+	case sdktypes.TriggerSourceTypeConnection:
+		sl.With("connection", trigger.ConnectionID()).Infof("creating connection trigger with connection %q", trigger.ConnectionID())
+	default:
+		return sdktypes.InvalidTriggerID, sdkerrors.NewInvalidArgumentError("unsupported source type")
 	}
 
-	// FIXME: see atomicity NOTE above
 	if err := m.db.CreateTrigger(ctx, trigger); err != nil {
 		return sdktypes.InvalidTriggerID, err
 	}
 
 	return trigger.ID(), nil
 }
 
-func (m *triggers) Update(ctx context.Context, trigger sdktypes.Trigger) error {
-	triggerID := trigger.ID()
-	prevTrigger, err := m.db.GetTrigger(ctx, triggerID)
+func (m *triggers) Update(ctx context.Context, next sdktypes.Trigger) error {
```

Comment:
what is next ?

Reply:
the new trigger data?

Reply:
The fact that this is not clear means you need to document it, or find a better self-documenting name.

Reply:
heh, i changed it because i thought it is more clear. oh well, fixed.